server/utils/risksAndNeeds/offenceAnalysisUtils.test.ts

```diff
@@ -0,0 +1,178 @@
+          },
+          value: {
+            text: 'Yes',
+          },
+        },
+      ])
+    })
+  })
+
+  describe('otherOffendersAndInfluencesSummaryListRows', () => {
+    it('formats the other offenders and influences offence details in the appropriate format for passing to a GOV.UK summary list Nunjucks macro', () => {
+      expect(OffenceAnalysisUtils.otherOffendersAndInfluencesSummaryListRows(offenceDetail)).toEqual([
+        {
+          key: { text: 'Were there other offenders involved?' },
+          value: {
+            text: 'Yes',
+          },
+        },
+        {
+          key: { text: 'Number of others involved' },
+          value: {
+            text: '2',
+          },
+        },
+        {
+          key: { text: 'Was the offender the leader?' },
+          value: {
+            text: 'Detail about the others involved',
+          },
+        },
+        {
+          key: { text: 'Peer group influences (e.g. offender easily led, gang member)' },
+          value: {
+            text: 'Detail about influences',
+          },
+        },
+      ])
+    })
+
+    describe('when there were no other offenders involved', () => {
+      it('returns "No" for the "Were there other offenders involved?" key', () => {
+        expect(
+          OffenceAnalysisUtils.otherOffendersAndInfluencesSummaryListRows(
+            offenceDetailFactory.build({ numberOfOthersInvolved: 0 }),
+          ),
+        ).toEqual(
+          expect.arrayContaining([
+            {
+              key: { text: 'Were there other offenders involved?' },
+              value: {
+                text: 'No',
+              },
+            },
+          ]),
+        )
+      })
+    })
+  })
+
+  describe('responsibilitySummaryListRows', () => {
+    it('formats the responsibilty related offence details in the appropriate format for passing to a GOV.UK summary list Nunjucks macro', () => {
+      expect(OffenceAnalysisUtils.responsibilitySummaryListRows(offenceDetail)).toEqual([
+        {
+          key: { text: 'Does the offender accept responsibility for the current offence(s)?' },
+          value: {
+            text: 'Yes',
+          },
+        },
+        {
+          key: {
+            text: 'How much responsibility does s/he acknowledge for the offence(s). Does s/he blame others, minimise the extent of his/her offending?',
+          },
+          value: {
+            text: 'Detail about the persons acceptance',
+          },
+        },
+      ])
+    })
+  })
+
+  describe('textValue', () => {
+    it('returns the text value', () => {
+      expect(OffenceAnalysisUtils.textValue('Some text')).toEqual('Some text')
+    })
+
+    describe('when the text value is null', () => {
+      it('returns the "No information available" message', () => {
+        expect(OffenceAnalysisUtils.textValue(null)).toEqual('No information available')
+      })
+    })
```

Comment:
I think it might become `undefined` on our end rather than `null`, or at least that's typically what we've tested for optional API fields